src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -108,24 +109,38 @@ private ResultSet executeQueryHelper(String querySql) throws SQLException {
 
         this.starttime = System.currentTimeMillis();
         Job referencedJob;
+        int retries = 0;
         // ANTLR Parsing
         BQQueryParser parser = new BQQueryParser(querySql, this.connection);
         querySql = parser.parse();
         try {
             // Gets the Job reference of the completed job with give Query
             referencedJob = startQuery(querySql);
         } catch (IOException e) {
-            throw new BQSQLException("Something went wrong with the query: " + querySql, e);
+            throw new BQSQLException("Something went wrong creating the query: " + querySql, e);
         }
         try {
             do {
                 if (this.connection.isClosed()) {
                     throw new BQSQLException("Connection is closed");
                 }
-                if (BQSupportFuncts.getQueryState(referencedJob,
-                        this.connection.getBigquery(),
-                        this.ProjectId.replace("__", ":").replace("_", ".")).equals(
-                        "DONE")) {
+
+                String status;
+                try {
+                    status = BQSupportFuncts.getQueryState(referencedJob,
+                            this.connection.getBigquery(),
+                            this.ProjectId.replace("__", ":").replace("_", "."));
+                } catch (IOException e) {
+                    if (retries++ < MAX_IO_FAILURE_RETRIES) {
+                       continue;
```

Comment:
- Should we sleep before retrying? Presumably the failure is from the serve being overloaded.
- Some debug logging might be useful.